extensions/amp-story/1.0/parallax-fx.js

```diff
@@ -0,0 +1,352 @@
+
+    this.init_();
+
+    userAssert(
+      this.win_.DeviceOrientationEvent,
+      "The current browser doesn't support the device motion/orientation events"
+    );
+
+    if (this.win_.DeviceOrientationEvent) {
+      listen(this.win_, 'deviceorientation', event => {
+        this.parallaxOrientationMutate_(event, this.parallaxPages_);
+      });
+    }
+  }
+
+  /**
+   * Initializes the pages by giving them perspective and initializating their layers
+   */
+  init_() {
+    this.vsync_.mutate(() => {
+      this.getPages()
+        .filter(page => !page.hasAttribute(NO_PARALLAX_FX_ATTR))
+        .forEach(page => {
+          const layers = this.getLayers(page);
+
+          // Set the page's perspective
+          setStyles(page, {
+            perspective: PERSPECTIVE * layers.length + `px`,
+          });
+
+          this.parallaxPages_.push(new ParallaxPage(page, this.vsync_));
+
+          const mode =
+            page.getAttribute(MODE_ATTR) || this.storyMode_ || DEFAULT_MODE;
+          const layerSpacing = Number(
+            page.getAttribute(LAYER_SPACING_ATTR) ||
+              this.storyLayerSpacing_ ||
+              DEFAULT_LAYER_SPACING
+          );
+          const farthestScale = Number(
+            page.getAttribute(FARTHEST_SCALE_ATTR) ||
+              this.storyFarthestScale_ ||
+              DEFAULT_FARTHEST_SCALE
+          );
+          const nearestScale = Number(
+            page.getAttribute(NEAREST_SCALE_ATTR) ||
+              this.storyNearestScale_ ||
+              DEFAULT_NEAREST_SCALE
+          );
+
+          this.initLayers_(
+            layers,
+            mode,
+            layerSpacing,
+            farthestScale,
+            nearestScale
+          );
+        });
+    });
+  }
+
+  /**
+   * Initializes the page's layers by giving them the appropriate translation and scaling
+   * @param {!Array<!Element>} layers
+   * @param {string} mode
+   * @param {number} layerSpacing
+   * @param {number} farthestScale
+   * @param {number} nearestScale
+   */
+  initLayers_(layers, mode, layerSpacing, farthestScale, nearestScale) {
+    // Loop through the layers in the page and assign a z-index following
+    // DOM order (manual override will be added in the future)
+    let order = 1;
+
+    const layerZIndexOffset =
+      mode == ParallaxModes.DEPTH
+        ? layers.length
+        : mode == ParallaxModes.CENTER
+        ? layers.length / 2
+        : 0;
+
+    layers.map(layer => {
+      this.vsync_.mutate(() => {
+        const translation = `translateZ(${(order - layerZIndexOffset) *
+          layerSpacing *
+          (PERSPECTIVE / 50)}px)`;
+        const scale = `scale(${mapRange(
+          order,
+          1,
+          layers.length,
+          farthestScale,
+          nearestScale
+        )})`;
+        setImportantStyles(layer, {
+          contain: 'none',
+          overflow: 'visible',
+          transform: `${translation} ${scale}`,
+        });
+        order++;
```

Comment:
The second argument to `forEach` is the index, no need to manually keep track of it yourself.